pkg/local/posix_permissions.go

```diff
@@ -68,8 +68,13 @@ func GetDefaultPermissions(isDir bool) POSIXPermissions {
 }
 
 // getPermissionFromStats - Get POSIX mode and ownership from object metadata, fallback to default permissions in case metadata doesn't exist
-func getPermissionFromStats(stats apigen.ObjectStats) (*POSIXPermissions, error) {
-	permissions := GetDefaultPermissions(strings.HasSuffix(stats.Path, uri.PathSeparator))
+// and setDefault is true
+func getPermissionFromStats(stats apigen.ObjectStats, setDefault bool) (*POSIXPermissions, error) {
+	permissions := POSIXPermissions{}
+	if setDefault {
+		permissions = GetDefaultPermissions(strings.HasSuffix(stats.Path, uri.PathSeparator))
+	}
+
 	if stats.Metadata != nil {
```

Comment:
When there's no default, it'll return empty `POSIXPermissions` if `stats.Metadata == nil`.
Is this by design?

Reply:
Yes, it's because when we compare between a local file and a remote object that still doesn't have permissions set we want to have the diff. Otherwise it can cause some inconsistent behavior